Remove `backtrace` example, move into docs instead (#2842)
boutproject/BOUT-dev: manual/sphinx/developer_docs/debugging.rst

    ====== Exception thrown ======
    Something went wrong

The third ``TRACE`` message doesn't appear in the output because we've
left its scope and it's no longer relevant.

The run-time overhead of this should be small, but can be removed
entirely if the compile-time flag ``-DCHECK`` is not defined or set to
``0``. This turns off checking, and ``TRACE`` becomes an empty
macro. This means that ``TRACE`` macros can be left in your code
permanently, providing some simple diagnostics without compromising
performance, as well as demarcating separate sections with
user-friendly names.

If you need to capture runtime information in the message, you can use
the ``fmt`` syntax also used by the loggers::

    TRACE("Value of i={}, some arbitrary {}", i, "string");

There is also an ``AUTO_TRACE`` macro that automatically captures the
name of the function it's used in. This is used throughout the main
library, especially in functions where numerical issues are likely to
arise.

Backtrace
=========

Lastly, BOUT++ can also automatically print a backtrace in the event
of a crash. This is a compile-time option in the BOUT++ library

> (``-DBOUT_ENABLE_BACKTRACE=on``, the default, requires the

Reviewer: ```suggestion
(``-DBOUT_ENABLE_BACKTRACE=ON``, the default, requires the
```

``addr2line`` program to be installed), and debug symbols to be turned
on (``-DCMAKE_BUILD_TYPE=Debug`` or ``=RelWithDebInfo``) in BOUT++

> _and_ the physics model.

Reviewer: ```suggestion
_and_ the physics model. If debug symbols are only present in part, the
backtrace will be missing names for the other part.
```

    [bt] #7 /path/to/BOUT-dev/build/libbout++.so(_ZN6Solver8setModelEP12PhysicsModel+0xb5) [0x7fecc0ca2e93]
    Solver::setModel(PhysicsModel*) at /path/to/BOUT-dev/build/../src/solver/solver.cxx:94
    [bt] #6 /path/to/BOUT-dev/build/libbout++.so(_ZN12PhysicsModel10initialiseEP6Solver+0xc0) [0x7fecc0cad594]
    PhysicsModel::initialise(Solver*) at /path/to/BOUT-dev/build/../include/bout/physicsmodel.hxx:93 (discriminator 5)
    [bt] #5 ./backtrace() [0x40a986]
    Backtrace::init(bool) at /path/to/BOUT-dev/build/../examples/backtrace/backtrace.cxx:27
    [bt] #4 ./backtrace() [0x40a3cf]
    f3() at /path/to/BOUT-dev/build/../examples/backtrace/backtrace.cxx:19
    [bt] #3 ./backtrace() [0x40a3be]
    f2(int) at /path/to/BOUT-dev/build/../examples/backtrace/backtrace.cxx:15
    [bt] #2 ./backtrace() [0x40a386]
    f1() at /path/to/BOUT-dev/build/../examples/backtrace/backtrace.cxx:13 (discriminator 2)
    [bt] #1 ./backtrace(_ZN13BoutExceptionC1IA19_cJEEERKT_DpRKT0_+0xba) [0x40ae16]
    BoutException::BoutException<char [19]>(char const (&) [19]) at /path/to/BOUT-dev/build/../include/bout/../boutexception.hxx:28 (discriminator 2)
              

> This output tends to be much less user-friendly than the message stack

Reviewer: ```suggestion
This output tends to be much harder to read than the message stack
```

I find it much friendlier to get in most cases all the info I need without changing the code :-)
